shiny/ui/_layout_columns.py

```diff
@@ -0,0 +1,305 @@
+
+    Returns
+    -------
+    :
+        An :class:`~htmltools.Tag` element.
+
+    See Also
+    --------
+    * :func:`~shiny.ui.layout_column_wrap` for laying out elements into a uniform grid.
+
+    Reference
+    --------
+    * [Bootstrap CSS Grid](https://getbootstrap.com/docs/5.3/layout/grid/)
+    * [Bootstrap Breakpoints](https://getbootstrap.com/docs/5.3/layout/breakpoints/)
+    """
+    attrs, children = consolidate_attrs(*args, class_=class_, **kwargs)
+
+    col_widths_spec = as_col_spec(col_widths, len(children))
+
+    # Create the bslib-layout-columns element
+    tag = Tag(
+        "bslib-layout-columns",
+        {
+            "class": "bslib-grid grid",
+            "style": css(
+                gap=as_css_unit(gap),
+                height=as_css_unit(height),
+            ),
+        },
+        col_widths_attrs(col_widths_spec),
+        attrs,
+        row_heights_attrs(row_heights),
+        *wrap_all_in_grid_item_container(children, fillable),
+        web_component_dependency(),
+    )
+
+    # Apply fill and fillable
+    if fill:
+        tag = as_fill_item(tag)
+    if fillable:
+        tag = as_fillable_container(tag)
+
+    return tag
+
+
+def wrap_all_in_grid_item_container(
+    children: Iterable[TagChild],
+    fillable: bool = True,
+    class_: Optional[str] = None,
+) -> list[TagChild]:
+    item_class = "bslib-grid-item bslib-gap-spacing"
+    if class_ is not None:
+        item_class = f"{item_class} {class_}"
+
+    # Use a new list so that we don't mutate the original `children`
+    wrapped_children: list[TagChild] = []
+    for child_value in children:
+        child = div({"class": item_class}, child_value)
+        if fillable:
+            child = as_fillable_container(child)
+        wrapped_children.append(child)
+
+    return wrapped_children
+
+
+def validate_col_width(
+    x: Iterable[int] | int, n_kids: int, break_name: str
+) -> Iterable[int]:
+    if isinstance(x, int):
+        y = [x]
+    else:
+        y = x
+
+    if not all(isinstance(i, int) for i in y):
+        raise ValueError(
+            f"Column values at breakpoint '{break_name}' must be integers. Values greater than 0 indicate width, and negative values indicate a column offset."
+        )
+
+    if any(i == 0 for i in y):
+        raise ValueError(
+            f"Column values at breakpoint '{break_name}' must be greater than 0 to indicate width, or negative to indicate a column offset."
+        )
+
+    if not any(b > 0 for b in y):
+        raise ValueError(
+            f"Column values at breakpoint '{break_name}' must include at least one positive integer width."
+        )
+
+    if len(list(y)) > n_kids:
+        warn(
+            f"More column widths than children at breakpoint '{break_name}', extra widths will be ignored."
+        )
+
+    return y
+
+
+def as_col_spec(
+    col_widths: BreakpointsUser[int],
+    n_kids: int,
+) -> BreakpointsOptional[int] | None:
```

Comment:
It'd be more readable for me if this func came immediately after `def layout_columns`